### PR TITLE
TRC-27- Add `Clickablelogo` component

### PR DESCRIPTION
**Pull request by mmcbride2929**
## TL/DR
This story creates an clickablelogo component story & component by combining the icon, and wordmark component.

## Overview of Changes

`ClickableLogo.tsx`
- Import Icon and Wordmark components 
- Created our ClickableLogo component by passing our icon component  an SVG, along with our wordmark component. 
- `Icon` & `Wordmark` were also wrapped with containers specified by classNames for styling. 
- The handleClick function was created and passed to our logo-container div.

`ClickableLogo.stories.tsx`
- Created clickable logo default story that can be used within a nav and/or footer

`ClickableLogo.scss`
- Style containers to match example

`redditWordmark.svg`
- Resize wordmark SVG to remove empty space around it

`Wordmark.scss`
- Default margin was removed from the h1 wrapper 

`Icon.tsx`
- Added optional customTestId prop to identify specific component for test

`Wordmark.tsx`
- Added optional customTestId prop to identify specific component for test

`ClickableLogo.test.tsx`
- Created 3 tests to make sure our component is functioning correctly.

### CODE REVIEW

File: apps/web/src/components/ClickableLogo/ClickableLogo.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.logo-container {
+  display: flex;
+  align-items: center;
+}
+
+.icon-container {
+  width: 35%;
+  padding: 25px;
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.scss:8]: Can you change these from `px` to a relative unit like `rem`?

File: apps/web/src/components/ClickableLogo/ClickableLogo.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+.wordmark-container {
+  width: 65%;
+  padding: 25px;
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.scss:13]: Same as above.

File: apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import ClickableLogo from './ClickableLogo';
+
+export default {
+  title: 'Components/ClickableLogo',
+  component: ClickableLogo,
+};
+
+export const Default = () => <ClickableLogo />;
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx:8]: Since we aren't using composition for this component can you use the `template.bind({})` syntax?

File: apps/web/src/components/ClickableLogo/ClickableLogo.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * @returns {React.ReactElement} A JSX element as a clickable reddit logo component.
+ */
+
+const ClickableLogo = (): React.ReactElement => {
+  /**
+   * handles click of logo
+   *
+   * @returns {string} confirming the logo has been clicked
+   *
+   * @description This function is called when the user clicks on the logo. It logs a message to the console to confirm that the click has been registered.
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.tsx:19]: Good detail here. Just as a note we can consider `@description` as optional.

File: apps/web/src/components/ClickableLogo/ClickableLogo.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+   */
+
+  const handleClick: () => void = () => {
+    console.log('Click Registered');
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.tsx:23]: Can this be refactored to call a callback that is passed as a prop to make this more flexible? Like what if we want to have it do one thing when clicked when its on the main page but have it do something else when its somewhere else?

**Review by wijohnst — COMMENTED** [review]
Great start! Left a few comments. Let me know if you have questions.

File: apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+export default {
+  title: 'Components/ClickableLogo',
+  component: ClickableLogo,
+};
+
+interface Props {
+  onClick: () => void;
+}
+
+const Template = ({ onClick }: Props) => <ClickableLogo onClick={onClick} />;
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx:12]: Remember, this is not the default way of defining a story. Unless we need to pass in a child component stories should always be defined using the `bind({})` syntax. 

And if you do it that way you don't need to define a special Props type. 

Also, if you ever need to use the `Props` type from a component outside of the component you should export the type from the component file and import it where you need it. If you do it the way you've done it here, you're creating an unnecessary dependency. What happens if we add a new `Prop` to `ClickableLogo`? You'd have to update it in `ClickabaleLogo.tsx` and here in the story. That could get messy.

File: apps/web/src/components/ClickableLogo/ClickableLogo.test.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { screen, render, fireEvent } from '@testing-library/react';
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.test.tsx:1]: Can you refactor to use `useEvent`?

Screenshot 2023-03-24 at 7 48 22 AM

https://testing-library.com/docs/user-event/intro

File: apps/web/src/components/Wordmark/Wordmark.tsx [hunk 8]
```diff
@@ -1,16 +1,23 @@
 import * as React from 'react';
+import './Wordmark.scss';
 import Icon, { iconColor } from '../Icon/Icon';
 import { ReactComponent as RedditWordmark } from '../../assets/redditWordmark.svg';
 
+interface Props {
+  customTestId?: string;
```
Comment (wijohnst) [inline apps/web/src/components/Wordmark/Wordmark.tsx:7]: Add JSDocs

**Review by wijohnst — COMMENTED** [review]
Great start! Left a few suggestions.

File: apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import * as React from 'react';
+import ClickableLogo from './ClickableLogo';
+
+export default {
+  title: 'Components/ClickableLogo',
+  component: ClickableLogo,
+  argTypes: {
+    onClick: { action: 'clicked' },
+  },
+};
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx:10]: You are missing the TypeScript for this story and that's why you have to define a `ClickableLogoArgs` type and why you're `Default.args` isn't working. 

So your stories should always have the same type as the component they represent, in this case `ClickableLogo.tsx`. But... out-of-the-box, `Storybook` doesn't know the types for `ClickableLogo`, so we've got to tell it. 

You're missing the `meta` and `storyFn` types that come from `Storybook`. Add those and you should be able to remove the `ClickableLogoArgs` type and everything should work.

File: apps/web/src/components/ClickableLogo/ClickableLogo.test.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  // testing for Wordmark component
+  it('should have a Wordmark component', () => {
+    render(<ClickableLogo onClick={handleClick} />);
+    expect(screen.getByTestId('clickableicon-wordmark')).toBeInTheDocument();
+  });
+
+  // testing that our handleClick fires
+  it('should trigger onClick function when the logo is clicked', async () => {
+    render(<ClickableLogo onClick={handleClick} />);
+    await userEvent.click(screen.getByTestId('clickablelogo'));
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.test.tsx:23]: Nice work on converting to `userEvent`.

File: apps/web/src/components/ClickableLogo/ClickableLogo.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import './ClickableLogo.scss';
+import Icon, { iconColor } from '../Icon/Icon';
+import Wordmark from '../Wordmark/Wordmark';
+import { ReactComponent as RedditIcon } from '../../assets/redditIcon.svg';
+
+interface Props {
+  /** What should happen with the component is clicked? */
+  onClick: () => void;
+  /** how can we identify our elements for testing?  */
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.tsx:10]: Nit:

```suggestion
  /** How can we identify our elements for testing?  */
```

File: apps/web/src/components/Icon/Icon.tsx [hunk 12]
```diff
@@ -7,8 +7,12 @@ export enum iconColor {
 }
 
 interface Props {
+  /** What should be displayed? */
   children: React.ReactElement;
+  /** The color of the Icon */
   colorProp?: iconColor;
+  /** What should be displayed? */
```
Comment (wijohnst) [inline apps/web/src/components/Icon/Icon.tsx:14]: NIT: This is the same comment as line 10

Also, the JSDocs don't _always_ need to be phrased as a question, especially if it doesn't make the comment clearer. Its a thing you'll get a feel for with time.

File: apps/web/src/components/Wordmark/Wordmark.tsx [hunk 13]
```diff
@@ -1,16 +1,24 @@
 import * as React from 'react';
+import './Wordmark.scss';
 import Icon, { iconColor } from '../Icon/Icon';
 import { ReactComponent as RedditWordmark } from '../../assets/redditWordmark.svg';
 
+interface Props {
+  /** how can we identify our elements for testing?  */
```
Comment (wijohnst) [inline apps/web/src/components/Wordmark/Wordmark.tsx:7]: NIT: Upperace
```suggestion
  /** How can we identify our elements for testing?  */
```

**Review by wijohnst — COMMENTED** [review]
Thanks for making those changes. Found a few more things.

File: apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,17 @@
+import * as React from 'react';
+import ClickableLogo, { Props } from './ClickableLogo';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+
+export default {
+  title: 'Components/ClickableLogo',
+  component: ClickableLogo,
+  argTypes: {
+    onClick: { action: 'clicked' },
+  },
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx:10]: Nice. Good use of `argTypes` to add control to the story.

File: apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+};
```
Comment (wijohnst) [inline apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx:11]: There should be additional typing here.
```suggestion
} as ComponentMeta<typeof ClickableLogo>; 
```

Comment (wijohnst) [reply apps/web/src/components/ClickableLogo/ClickableLogo.stories.tsx:11]: For more on the `as` type assertion in TypeScript see:

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions

**Review by wijohnst — COMMENTED** [review]
Left one final comment to address.

**Review by wijohnst — APPROVED** [review]
LGTM!
